sites-enabled/11-api.freecodecamp.org.conf

```
@@ -20,6 +20,9 @@ server {
       return 503;
     }
     proxy_pass http://api;
+    proxy_set_header X-Forwarded-For CF-Connecting-IP;
```

Hey, I did a little digging and I think we want $remote_addr here, since that gets set to the value of CF-Connecting-IP.

I've not had chance to test this, but I'll report back when I have.

---

I wanted it to be more in the face for future maintenance. 

I know we can use `$remote_addr` and swap that in the future if we move. But I prefer this slightly more when I am auditing the logic later: "API depends on the CF-Connecting-IP"

That is, say there is a greater chance of going to Caddy for example; I would prefer looking at the location blocks than the abstractions we have for configs (which, BTW is one of the regrets we have as tech debt)

---

Okay, that makes sense. In that case I believe
```suggestion
    proxy_set_header X-Forwarded-For $http_cf_connecting_ip;
```
is what we want. "CF-Connecting-IP" is just text.

---

I'd like to test this in staging, though, to be 100% sure. Let me know when I can do that without getting in your way.

---

This exact code was deployed and tested yesterday and is from their docs.

---

I know, I saw the db entries you posted in chat, though. The keys stored in the rate-limiting collection are not what we intended. They should be the real ips, but they're not.

What's happening is `proxy_set_header X-Forwarded-For CF-Connecting-IP;` sets the XFF header to the string `CF-Connecting-IP`, not the value of the header `CF-Connecting-IP`.

We also have

https://github.com/freeCodeCamp/nginx-config/blob/124da11b74ed7396dce4627e8f5943ecfe55169a/snippets/common/proxy-params.conf#L8

So, we overwrite the XFF header for a second time and end up with the keys you shared in chat.